src/app/components/student/student.component.ts

```diff
@@ -171,7 +168,7 @@ export class StudentComponent implements OnInit {
     }
   }
 
-  setQuery(filterArray: Array<FilterField>): string {
+  getQuery(filterArray: Array<FilterModel>): string {
```

Comment:
It is not clear what query we are retrieving here, maybe better naming would be 'getFilterQuery'?